[Java] class not found CLASSNAME:demo.fury.pkg1.Struct loader:jdk.internal.loader.ClassLoaders$AppClassLoader

Thank you for reporting the problem!
Please make sure what you are reporting is a bug with reproducible steps. 

**Describe the bug**
FST/kryo benchmark can't deserialzie objects of dynamic generated classes.
```
java.io.IOException: java.lang.RuntimeException: class not found CLASSNAME:demo.fury.pkg1.Struct loader:jdk.internal.loader.ClassLoaders$AppClassLoader@4b85612c
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:247)
        at org.nustaq.serialization.FSTConfiguration.asObject(FSTConfiguration.java:1158)
        at io.fury.benchmark.state.FstState$FstBenchmarkState.deserialize(FstState.java:119)
        at io.fury.benchmark.state.FstState$FstBenchmarkState.deserialize(FstState.java:104)
        at io.fury.benchmark.state.FstState$FstUserTypeState.setup(FstState.java:173)
        at io.fury.benchmark.jmh_generated.UserTypeSerializeSuite_fst_serialize_jmhTest._jmh_tryInit_f_fstusertypestate1_1(UserTypeSerializeSuite_fst_serialize_jmhTest.java:353)
        at io.fury.benchmark.jmh_generated.UserTypeSerializeSuite_fst_serialize_jmhTest.fst_serialize_Throughput(UserTypeSerializeSuite_fst_serialize_jmhTest.java:73)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.RuntimeException: class not found CLASSNAME:demo.fury.pkg1.Struct loader:jdk.internal.loader.ClassLoaders$AppClassLoader@4b85612c
        at org.nustaq.serialization.FSTClazzNameRegistry.classForName(FSTClazzNameRegistry.java:235)
        at org.nustaq.serialization.FSTClazzNameRegistry.classForName(FSTClazzNameRegistry.java:190)
        at org.nustaq.serialization.FSTClazzNameRegistry.decodeClass(FSTClazzNameRegistry.java:173)
        at org.nustaq.serialization.coders.FSTStreamDecoder.readClass(FSTStreamDecoder.java:478)
        at org.nustaq.serialization.FSTObjectInput.readClass(FSTObjectInput.java:939)
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:347)
        at org.nustaq.serialization.FSTObjectInput.readObjectInternal(FSTObjectInput.java:331)
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:311)
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:245)
```
```
com.esotericsoftware.kryo.KryoException: Unable to find class: demo.fury.pkg1.Struct
        at com.esotericsoftware.kryo.util.DefaultClassResolver.readName(DefaultClassResolver.java:160)
        at com.esotericsoftware.kryo.util.DefaultClassResolver.readClass(DefaultClassResolver.java:133)
        at com.esotericsoftware.kryo.Kryo.readClass(Kryo.java:693)
        at com.esotericsoftware.kryo.Kryo.readClassAndObject(Kryo.java:804)
        at io.fury.benchmark.state.KryoState$KryoUserTypeState.setup(KryoState.java:146)
        at io.fury.benchmark.jmh_generated.UserTypeDeserializeSuite_kryo_deserialize_compatible_jmhTest._jmh_tryInit_f_kryocompatiblestate1_1(UserTypeDeserializeSuite_kryo_deserialize_compatible_jmhTest.java:353)
        at io.fury.benchmark.jmh_generated.UserTypeDeserializeSuite_kryo_deserialize_compatible_jmhTest.kryo_deserialize_compatible_Throughput(UserTypeDeserializeSuite_kryo_deserialize_compatible_jmhTest.java:73)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: demo.fury.pkg1.Struct
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at com.esotericsoftware.kryo.util.DefaultClassResolver.readName(DefaultClassResolver.java:154)
        ... 18 more
```
**Expected behavior**
A clear and concise description of what you expected to happen.

**To Reproduce**
run fury-benchmark

**Screenshots**
![image](https://github.com/alipay/fury/assets/12445254/8f5cf31e-762e-4d2b-aa0a-b8eb1706ce94)
![image](https://github.com/alipay/fury/assets/12445254/f6bac598-00c7-422a-bcca-f565464db9ae)


**Environment (please complete the following information):**
Please specify the versions of Fury, OS, and others such as JDK/Python/Gcc/Clang/Go/NodeJS if they are used.

--- a/java/fury-benchmark/runtest_jdk11.sh
+++ b/java/fury-benchmark/runtest_jdk11.sh
@@ -15,4 +15,4 @@ done
 echo "Oracle JDK11 Serialization: ZeroCopySuite"
 oracle_jdk 11
 REGISTER_CLASS=false java -jar target/benchmarks.jar "io.*\.ZeroCopySuite\.*" -f 1 -wi 3 -i 3 -t 1 -w 2s -r 2s -rf csv
-cp jmh-result.csv jmh-result-jdk-11-false-zerocopy-deserialization.csv
+cp jmh-result.csv jmh-result-jdk-11-false-zerocopy.csv

--- a/java/fury-benchmark/src/main/java/io/fury/benchmark/UserTypeDeserializeSuite.java
+++ b/java/fury-benchmark/src/main/java/io/fury/benchmark/UserTypeDeserializeSuite.java
@@ -112,7 +112,7 @@ public class UserTypeDeserializeSuite {
   public static void main(String[] args) throws IOException {
     if (args.length == 0) {
       String commandLine =
-          "io.*UserTypeDeserializeSuite.furymetashared_deserialize_compatible -f 0 -wi 3 -i 3 -t 1 -w 2s -r 2s -rf csv";
+          "io.*UserTypeDeserializeSuite.kryo* -f 0 -wi 3 -i 3 -t 1 -w 2s -r 2s -rf csv";
       System.out.println(commandLine);
       args = commandLine.split(" ");
     }

--- a/java/fury-benchmark/src/main/java/io/fury/benchmark/state/FstState.java
+++ b/java/fury-benchmark/src/main/java/io/fury/benchmark/state/FstState.java
@@ -27,8 +27,6 @@ import io.fury.benchmark.data.Data;
 import io.fury.benchmark.data.Image;
 import io.fury.benchmark.data.Media;
 import io.fury.benchmark.data.MediaContent;
-import io.fury.benchmark.data.Sample;
-import io.fury.benchmark.data.Struct;
 import io.fury.util.Platform;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
@@ -68,12 +66,16 @@ public class FstState {
 
     @Setup(Level.Trial)
     public void setup() {
-      fst = FSTConfiguration.createDefaultConfiguration();
-      fst.setPreferSpeed(true);
-      fst.setShareReferences(references);
+      setupFst();
       if (bufferType == BufferType.directBuffer) {
         directBuffer = ByteBuffer.allocateDirect(10 * 1024 * 1024);
       }
+    }
+
+    public void setupFst() {
+      fst = FSTConfiguration.createDefaultConfiguration();
+      fst.setPreferSpeed(true);
+      fst.setShareReferences(references);
     }
 
     public static byte[] serialize(Blackhole blackhole, FstBenchmarkState state, Object value) {
@@ -133,20 +135,20 @@ public class FstState {
 
     @Override
     public void setup() {
-      if (objectType == ObjectType.STRUCT) {
-        Thread.currentThread()
-            .setContextClassLoader(Struct.createStructClass(110, false).getClassLoader());
-      }
+      object = ObjectType.createObject(objectType, references);
+      Thread.currentThread().setContextClassLoader(object.getClass().getClassLoader());
       super.setup();
+      setupFst();
+      fst.setClassLoader(object.getClass().getClassLoader());
       switch (objectType) {
         case SAMPLE:
-          object = new Sample().populate(references);
+        case STRUCT2:
+        case STRUCT:
           if (registerClass) {
             fst.registerClass(object.getClass());
           }
           break;
         case MEDIA_CONTENT:
-          object = new MediaContent().populate(references);
           if (registerClass) {
             fst.registerClass(Image.class);
             fst.registerClass(Image.Size.class);
@@ -156,18 +158,8 @@ public class FstState {
             fst.registerClass(MediaContent.class);
           }
           break;
-        case STRUCT:
-          object = Struct.create(false);
-          if (registerClass) {
-            fst.registerClass(object.getClass());
-          }
-          break;
-        case STRUCT2:
-          object = Struct.create(true);
-          if (registerClass) {
-            fst.registerClass(object.getClass());
-          }
-          break;
+        default:
+          throw new IllegalStateException("Unexpected value: " + objectType);
       }
       buffer = serialize(null, this, object);
       Preconditions.checkArgument(object.equals(deserialize(null, this)));

--- a/java/fury-benchmark/src/main/java/io/fury/benchmark/state/KryoState.java
+++ b/java/fury-benchmark/src/main/java/io/fury/benchmark/state/KryoState.java
@@ -97,6 +97,7 @@ public class KryoState {
       object = ObjectType.createObject(objectType, references);
       Thread.currentThread().setContextClassLoader(object.getClass().getClassLoader());
       kryo = new Kryo();
+      kryo.setClassLoader(object.getClass().getClassLoader());
       kryo.setReferences(references);
       if (compatible()) {
         kryo.setDefaultSerializer(CompatibleFieldSerializer.class);

